src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -325,11 +335,31 @@ def _forward(self, model_inputs):
                 # Send stride to `postprocess`.
                 # it needs to be handled there where
                 # the pieces are to be concatenated.
-                ratio = 1 / self.model.config.inputs_to_logits_ratio
+                if input_to_features_ratio is None:
+                    ratio = self.model.config.inputs_to_logits_ratio
+                else:
+                    new_ratio = inputs.shape[1] / logits.shape[1]
+                    if isinstance(input_to_features_ratio, list):
```

Comment:
Every single element of the batch will use the same ratio, and that's guaranteed.

The reason is that we're using the same `feature_extractor` (first ratio) and the same model (second ratio).
The exact numbers might vary because the size of the chunks could vary and not be aligned with said ratio (like 74 400 samples vs 74 350 sampling points).